msu/utils/text.nut

```diff
@@ -28,4 +28,97 @@
 	{
 		return this.color(::Const.UI.Color.NegativeValue, _string);
 	}
+
+	function colorizeValue( _value, _kwargs = null )
+	{
+		local kwargs = {
+			AddSign = true,
+			CompareTo = 0,
+			Invert = false,
+			AddPercent = false
+		};
+
+		if (_kwargs != null)
+		{
+			foreach (key, value in _kwargs)
+			{
+				if (!(key in kwargs)) throw "invalid parameter " + key;
+				kwargs[key] = value;
+			}
+		}
+
+		if (_value < kwargs.CompareTo)
+		{
+			if (!kwargs.AddSign && _value < 0) _value *= -1;
+			if (kwargs.AddPercent) _value = _value + "%";
+			return kwargs.Invert ? this.colorGreen(_value) : this.colorRed(_value);
+		}
+
+		if (_value > kwargs.CompareTo)
+		{
+			if (kwargs.AddSign && _value > 0) _value = "+" + _value;
+			if (kwargs.AddPercent) _value = _value + "%";
+			return kwargs.Invert ? this.colorRed(_value) : this.colorGreen(_value);
+		}
+
+		if (_value == kwargs.CompareTo)
+		{
+			if (kwargs.AddPercent) _value = _value + "%";
+			return _value;
+		}
+	}
+
+	// Returns 0.75 as red 25% and 1.75 as green 75%
+	// Used to colorize mults which are applied to other values
+	// Use case example: when someone has a 0.75 multiplier to their defense and you want to write "25% less defense" in tooltip
+	function colorizeMult( _value, _kwargs = null )
+	{
+		local kwargs = {
+			AddMoreLess = false,
```

Comment:
Should this default to `true` or `false`? In most of the practical use cases you almost always want to append `more` or `less`. But the "purity" of the function would suggest that it should only return a colorized mult by default without additional appendages.

Reply:
I like it being false. That's more intuitive for a first-time user and for modder who prefer other terms.

Reply:
I like it being false. That's more intuitive for a first-time user and for modder who prefer other terms.

Reply:
An alternative would be to pass an array of preferred words e.g. `MoreLess = ["more", "less"]`. By default we could have an empty array but this could allow modders to very easily use the same function but with custom terminology.

Reply:
I went with an alternative approach and added a separate function for colorizing mults with text. This keeps the overall thing simpler.